subprojects/core/src/main/java/org/gradle/cache/internal/DefaultGeneratedGradleJarCache.java

```diff
@@ -83,4 +64,9 @@ public void close() {
     private File jarFile(String identifier) {
         return new File(cache.getBaseDir(), "gradle-" + identifier + "-" + gradleVersion + ".jar");
     }
+
+    @Override
+    public List<File> getFileStoreRoots() {
+        return Collections.singletonList(cache.getBaseDir());
```

Comment:
❓This directory also includes jar generated for the Kotlin DSL, e.g. `gradle-kotlin-dsl-extensions-0.17.4-4.9-20180516185546+0000.jar` or `gradle-ktlint-convention-ruleset-f3f6536d6dad773b6d3406d8108510aa-4.9-20180516185546+0000.jar`. Can these also assumed to be immutable?

Reply:
Yes.